app/Console/Commands/CheckCommand.php

```diff
@@ -46,15 +47,20 @@ protected function configure(): void
             InputOption::VALUE_IS_ARRAY | InputOption::VALUE_REQUIRED,
             'Class types that should be skipped'
         );
+        $this->addOption(
+            'configuration',
+            'c',
+            InputOption::VALUE_REQUIRED,
+            'Configuration file name'
+        );
```

Comment:
Sure... written on the phone :)

```suggestion
     , getcwd() . '/class-leak.yaml'
        );
```

Reply:
Ok that's what I understood, but it's not that simple.
Because if this default config file does not exist it must not fail (as you said it is expected to be optional) whereas if it is a user specific config file it is expected to fail if it does not exist.

So what you want is this mostly standard fallback chain:
- load config file specified via -c
- or load default config file if it exists

And IMO it should be done in a separate patch.

Reply:
If you can create a PR to target this PR, go for it :) I want to merge together to have the expected behavior in main in one instant.

Reply:
I've implemented this basic fallback chain right in this PR.

Reply:
@TomasVotruba is it ok for you ?

Reply:
CI has to pass first. Please also use this suggestion, as standard for Symfony commands.

Protip: run commands locally, before you push:

```bash
vendor/bin/phpstan
vendor/bin/rector
```

Reply:
@TomasVotruba the default value is not compatible with the standard fallback chain I've described, because if we specifiy a default value representing an optional file, then we cannot fail if the file does not exist. This is substandard since it would hide an important error.
Could you give me a link of a Symfony command with the behavior you have in mind ?

Reply:
Instead of checking this, the goal is to allow default autoload so people won't have to define it.

The same way other packages do not define `bin/command -c default-path`

Reply:
Did you see the last commit which adds precisely the default config (`class-leak.yaml`) fallback ? https://github.com/TomasVotruba/class-leak/pull/9/commits/5dfe73e6d45e43d8e2dcdf042f31aa6c8350620b

Reply:
Yes, this should be directly in command option default as suggested.

Reply:
@TomasVotruba can you explain why it should be implemented with an option's default value ? And can you add a link of a "standard" Symfony command's source code which implements what you have in mind ?